pylint/checkers/base.py

```diff
@@ -2408,18 +2408,50 @@ def _is_singleton_const(node) -> bool:
     def _check_NaN_comparison(
         self, left_value, right_value, root_node, checking_for_absence: bool = False
     ):
-        nan_values = ("float('nan')", "numpy.NaN", "np.NaN")
+        def _is_float_nan(node):
+            try:
+                if (
+                    isinstance(node, astroid.Call)
+                    and node.args[0].value == "nan"
+                    and node.inferred()[0].pytype() == "builtins.float"
+                ):
+                    return True
+                return False
+            except (AttributeError, IndexError):
+                return False
+
+        def _is_numpy_NaN(node):
```

Comment:
Would it be possible to have uniform case for `nan` word?

Reply:
like `_is_nan_word` method that covers all the cases for `nan`?
I could do that.

Reply:
I am not sure to understand your answer. I just wanted to say that it could be nice to have the word `nan` (in lowercase) or `NaN` in all method names.

Reply:
Now I understand what you meant, will fix it